tensorboard/webapp/feature_flag/actions/feature_flag_actions.ts

```diff
@@ -43,3 +43,16 @@ export const featureFlagOverrideChanged = createAction(
     flags: Partial<FeatureFlags>;
   }>()
 );
+
+export const featureFlagOverridesReset = createAction(
```

Comment:
I'm a little suspicious about featureFlagOverrideChanged and featureFlagOverridesReset living side x side. It either points to something wrong with your UI or something wrong with how you're modeling actions.

I am guessing that:
* "featureFlagOverrideChanged" is fired by clicking on some sort of "Save" button at the bottom of the dialog.
* "featureFlagsOverridesAllSetToDefault" is fired by clickign on some sort of "Reset All" button at the bottom of the dialog.

Is that right? If so, where does "featureFlagOverridesReset" fit in? Is there a third "Reset Selected" button or similar? That doesn't seem quite right.

Reply:
featureFlagOverrideChanged is used when the user sets a flag to override as true or as false. However, the Partial<FeatureFlags> is not able to indicate that a flag is set to default. So featureFlagOverridesReset handles the setting to default.

I was under the impression we would save these individually as the user changes the setting(ie there is no "save" button). However, if we do have a save button then featureFlagOverrideChanged(which should be named featureFlagOverrideSaved in this case) should be all that is necessary and the flags that are missing from the Partial<FeatureFlags> object can all be set to default.

I think I prefer the option where we call these actions on individual changes and we do not require a save button.

Reply:
For a bit of background I wanted to make featureFlagOverrideChanged have a single key and value pair but I could not get the type to work the way I wanted. We decided to use Partial<FeatureFlags> as a compromise.

Reply:
Thanks for the explanation, that makes sense. Yes agreed that it would be fine/preferable to omit the "Save" button.

I assumed there is a "Save" button because these actions specify lists of flags. Should they instead specify a single flag?

```
props<{
  flag: keyof FeatureFlags
}>()
```

Reply:
I was trying to keep my new actions consistent with the existing ones. I believe the logic for using `Partial<FeatureFlags>` originally was to persist the key and value of the flag which lead to otherwise complicated typing.

Reply:
I am ok with this change if you feel strongly about it. We had discussed this as an option. The problem is that there is no way to guarantee the FeatureFlagType(which is basically a big OR statement with all the possible types) matches the type associated with the given key. It was all a lot cleaner and more strongly typed when we just used the Partial. I see the confusion this causes but, I still have a slight preference to the way it is now.

Reply:
Yes, FeatureFlagType is unfortunately necessary to workaround some (relatively minor) limitations in the TypeScript type system.

I think in this case we should prefer the more-accurate data model over more-accurate type checks.